File: PerfTest2/PerfTest2-Xamarin/PerfTest2-ClassicXamarin.Android/MainActivity.cs
using System;
using Android.App;
using Android.OS;
using Android.Support.V7.App;
using Android.Views;
using Android.Widget;
using PerfTest2Xamarin.Adapters;
using PerfTest2Xamarin.Enums;
using PerfTest2Xamarin.Fragments;
using PerfTest2Xamarin.Utilities;

namespace PerfTest2Xamarin
{
	[Activity (Label = "PerfTest2_ClassicXamarin.Android", MainLauncher = true, Icon = "@drawable/icon", Theme = "@style/AppTheme")]
	public class MainActivity : AppCompatActivity, AdapterView.IOnItemClickListener
	{
		Fragment currentFragment;
		private string directory;

		protected override void OnCreate (Bundle savedInstanceState)
		{
			base.OnCreate (savedInstanceState);

			// Set our view from the "main" layout resource
			SetContentView (Resource.Layout.activity_main);

			currentFragment = new MainMenuFragment ();
			this.FragmentManager.BeginTransaction ()
								.Add (Resource.Id.main_area, currentFragment)
								.AddToBackStack (null)
								.Commit ();
		}

		protected override void OnResume ()
		{
			base.OnResume ();

			directory = this.GetExternalFilesDir (Android.OS.Environment.DirectoryDownloads).Path + "/";
		}

		public void OnItemClick (AdapterView parent, View view, int position, long id)
		{
			var lstMainMenu = (ListView)this.FindViewById (Resource.Id.lstMainMenu);

			switch (position)
			{
			case MainMenuAdapter.CLEAN_UP_TEST:
				CleanUp ();
				break;
			case MainMenuAdapter.ADD_SQL_RECORDS:
				AddRecords ();
				break;
			case MainMenuAdapter.DISPLAY_ALL_RECORDS:
				ShowAllRecords ();
				break;
			case MainMenuAdapter.DISPLAY_RECORDS_WITH_WHERE:
				ShowRecordsWith ();
				break;
			case MainMenuAdapter.SAVE_LARGE_FILE:
				SaveLargeFile ();
				break;
			case MainMenuAdapter.DISPLAY_LARGE_FILE:
				LoadAndDisplayFile ();
				break;
			}
		}

		private void CleanUp ()
		{
			var sqlUtilities = new SqLiteUtilitiesAlt (this);
			var alertDialog = new Android.Support.V7.App.AlertDialog.Builder (this);
			try
			{
				sqlUtilities.CreateTable ();
				sqlUtilities.CloseConnection ();

				using (var fUtilities = new FileUtilities (directory))
				{
					fUtilities.DeleteFile ();
					fUtilities.CreateFile ();
					fUtilities.CloseFile ();
				}

				alertDialog.SetMessage ("Completed test setup");
				alertDialog.SetTitle ("Cleanup and Prepare for Tests Successful");
			}
			catch (Exception ex)
			{
				alertDialog.SetMessage ("An error has occurred: " + ex.Message);
				alertDialog.SetTitle ("Error");
			}
			finally
			{
				alertDialog.SetPositiveButton ("OK", (sender, args) => { });
				alertDialog.SetCancelable (true);
				alertDialog.Create ().Show ();
			}
		}

		private void AddRecords ()
		{
			var utilities = new SqLiteUtilitiesAlt (this);
			var alertDialog = new Android.Support.V7.App.AlertDialog.Builder (this);
			try
			{
				for (int i = 0; i <= 999; i++)
				{
					utilities.AddRecord ("test", "person", i, "12345678901234567890123456789012345678901234567890");
				}
				utilities.CloseConnection ();

				alertDialog.SetMessage ("All records written to database");
				alertDialog.SetTitle ("Success");
			}
			catch (Exception ex)
			{
				alertDialog.SetMessage ("An error has occurred adding records: " + ex.Message);
				alertDialog.SetTitle ("Error");
			}
			finally
			{
				alertDialog.SetPositiveButton ("OK", (sender, args) => { });
				alertDialog.SetCancelable (true);
				alertDialog.Create ().Show ();
			}
		}

		private void ShowAllRecords ()
		{
			var bundle = new Bundle ();
			bundle.PutInt ("displayType", (int)SqLiteDisplayType.ShowAll);
			var fragment = new SqLiteTableFragment (directory) {
				Arguments = bundle
			};
			FragmentManager.BeginTransaction ()
						   .Replace (Resource.Id.main_area, fragment)
						   .AddToBackStack (null)
						   .Commit ();
		}

		private void ShowRecordsWith ()
		{
			var bundle = new Bundle ();
			bundle.PutInt ("displayType", (int)SqLiteDisplayType.ShowContaining1);
			var fragment = new SqLiteTableFragment (directory) {
				Arguments = bundle
			};
			FragmentManager.BeginTransaction ()
						   .Replace (Resource.Id.main_area, fragment)
						   .AddToBackStack (null)
						   .Commit ();
		}

		private void SaveLargeFile ()
		{
			var alertDialog = new Android.Support.V7.App.AlertDialog.Builder (this);
			try
			{
				using (var utilities = new FileUtilities (directory))
				{
					for (int i = 0; i <= 999; i++)
					{
						utilities.WriteLineToFile ("Writing line to file at index: " + i);
					}
					utilities.CloseFile ();
				}

				alertDialog.SetMessage ("All lines written to file");
				alertDialog.SetTitle ("Success");
			}
			catch (Exception ex)
			{
				alertDialog.SetMessage ("An error has occurred adding lines to file: " + ex.Message);
				alertDialog.SetTitle ("Error");
			}
			finally
			{
				alertDialog.SetPositiveButton ("OK", (sender, args) => { });
				alertDialog.SetCancelable (true);
				alertDialog.Create ().Show ();
			}
		}

		private void LoadAndDisplayFile ()
		{
			var fragment = new DisplayTextFileFragment (directory);
			FragmentManager.BeginTransaction ()
						   .Replace (Resource.Id.main_area, fragment)
						   .AddToBackStack (null)
						   .Commit ();
		}
	}
}

File: PerfTest2/PerfTest2-Java/app/src/main/java/com/tamasszadvari/perftest2_java/MainActivity.java
package com.tamasszadvari.perftest2_java;

import android.app.AlertDialog;
import android.app.Fragment;
import android.app.FragmentManager;
import android.app.FragmentTransaction;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

public class MainActivity extends AppCompatActivity implements AdapterView.OnItemClickListener, SqLiteTableFragment.OnFragmentInteractionListener, DisplayTextFileFragment.OnFragmentInteractionListener {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Fragment currentFragment = new MainMenuFragment();
        getFragmentManager().beginTransaction()
                            .add(R.id.main_area,currentFragment)
                            .addToBackStack(null)
                            .commit();
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        ListView lstMainMenu = (ListView)this.findViewById(R.id.lstMainMenu);
        MainMenuAdapter adapter = (MainMenuAdapter)lstMainMenu.getAdapter();

        if (position == adapter.CLEAN_UP_TEST) {
            cleanUp();
        } else if (position == adapter.ADD_SQL_RECORDS) {
            addRecords();
        } else if (position == adapter.DISPLAY_ALL_RECORDS) {
            showAllRecords();
        } else if (position == adapter.DISPLAY_RECORDS_WITH_WHERE) {
            showRecordsWith();
        } else if (position == adapter.SAVE_LARGE_FILE) {
            saveLargeFile();
        } else if (position == adapter.DISPLAY_LARGE_FILE) {
            loadAndDisplayFile();
        }
    }

    private void cleanUp() {
        SqLiteUtilities sqlUtilities = new SqLiteUtilities(this);
        FileUtilities fUtilities = new FileUtilities(this);
        AlertDialog.Builder alert  = new AlertDialog.Builder(this);

        try {
            sqlUtilities.openConnection();
            sqlUtilities.createTable();
            sqlUtilities.closeConnection();

            fUtilities.deleteFile();
            fUtilities.createFile();

            alert.setMessage("Completed test setup");
            alert.setTitle("Cleanup and Prepare for Tests Successful");
        } catch (Exception ex) {
            alert.setMessage("An error has occurred: " + ex.getMessage());
            alert.setTitle("Error");
        } finally {
            alert.setPositiveButton("OK", null);
            alert.setCancelable(true);
            alert.create().show();
        }
    }

    private void addRecords() {
        SqLiteUtilities utilities = new SqLiteUtilities(this);
        AlertDialog.Builder alert  = new AlertDialog.Builder(this);

        try {
            for (int i = 0; i <= 999; i++) {
                utilities.addRecord("test", "person", i, "12345678901234567890123456789012345678901234567890");
            }

            alert.setMessage("All records written to database");
            alert.setTitle("Success");
        } catch (Exception ex) {
            alert.setMessage("An error has occurred adding records: " + ex.getMessage());
            alert.setTitle("Error");
        } finally {
            alert.setPositiveButton("OK", null);
            alert.setCancelable(true);
            alert.create().show();
        }
    }

    private void showAllRecords() {
        Bundle bundle = new Bundle();
        bundle.putSerializable("displayType", SqLiteDisplayType.ShowAll);

        SqLiteTableFragment fragment = new SqLiteTableFragment();
        fragment.setArguments(bundle);

        getFragmentManager().beginTransaction()
                            .replace(R.id.main_area,fragment)
                            .addToBackStack(null)
                            .commit();
    }

    private void showRecordsWith() {
        Bundle bundle = new Bundle();
        bundle.putSerializable("displayType", SqLiteDisplayType.ShowContaining1);

        SqLiteTableFragment fragment = new SqLiteTableFragment();
        fragment.setArguments(bundle);

        getFragmentManager().beginTransaction()
                            .replace(R.id.main_area,fragment)
                            .addToBackStack(null)
                            .commit();
    }

    private  void saveLargeFile() {
        FileUtilities utilities = new FileUtilities(this);
        AlertDialog.Builder alert  = new AlertDialog.Builder(this);

        try {
           for (int i = 0; i <= 999; i++) {
                utilities.writeLineToFile("Writing line to file at index: " + i + System.getProperty ("line.separator"));
            }

            utilities.closeFile();

            alert.setMessage("All lines written to file");
            alert.setTitle("Success");
        } catch (Exception ex) {
            alert.setMessage("An error has occurred adding lines to file: " + ex.getMessage());
            alert.setTitle("Error");
        } finally {
            alert.setPositiveButton("OK", null);
            alert.setCancelable(true);
            alert.create().show();
        }
    }

    private void loadAndDisplayFile() {
        DisplayTextFileFragment fragment = new DisplayTextFileFragment();

        getFragmentManager().beginTransaction()
                            .replace(R.id.main_area,fragment)
                            .addToBackStack(null)
                            .commit();
    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }
}
